src/tests/testsuite.ts

```diff
@@ -112,7 +113,7 @@ export const createTestSuite: CreateTestsuite = (name, doubleCsrfOptions) => {
         assert.notEqual(generatedToken, csrfToken);
       });
 
-      it("should throw if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value) and overwrite is false", () => {
+      it("should throw if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value), overwrite is false, and validateOnGeneration is true", () => {
```

Comment:
should throw if csrf cookie is present and invalid, overwrite is false, and validateOnGeneration is enabled